crates/rspack_core/src/compiler/compilation.rs

```diff
@@ -1970,10 +1969,15 @@ impl Compilation {
       .chunk_graph
       .connect_chunk_and_runtime_module(*chunk_ukey, runtime_module_identifier);
 
-    self
+    if let Some(new_source) = self
       .plugin_driver
-      .runtime_module(module.as_mut())
-      .expect("Run runtime_module hook failed");
+      .clone()
```

Comment:
Can we avoid cloning here? not passing `self` down to the hook

Reply:
Fixed cloning. Currently runtime modules need the full compilation to generate their codes. So `self` here need to be passed.